src/connection-manager/manifests.ts

```diff
@@ -244,16 +297,48 @@ async function updateRevocations(): Promise<void> {
 }
 
 /**
- * Deletes validators that are older than an hour.
+ * Deletes validators that are older than a week, but keeps validators that are in the UNL.
  *
  * @returns Void.
  */
-async function purgeOldValidators(): Promise<void> {
+export async function purgeOldValidators(): Promise<void> {
   const oneWeekAgo = new Date()
   oneWeekAgo.setDate(oneWeekAgo.getDate() - 7)
   log.info('Deleting old validators')
   try {
-    await query('validators').where('last_ledger_time', '<', oneWeekAgo).del()
+    // Get all UNL validators from all networks
+    const networks = (await getNetworks()).map((network) => network.id)
+    const unlValidatorSigningKeys: Set<string> = new Set()
+
+    for (const network of networks) {
+      try {
+        const unl: UNLBlob = await fetchValidatorList(
+          await getFirstUNL(network),
+        )
+        unl.validators.forEach((validator: UNLValidator) => {
+          const manifestHex = Buffer.from(validator.manifest, 'base64')
+            .toString('hex')
+            .toUpperCase()
+          const manifest = normalizeManifest(manifestHex)
+          if (manifest.signing_key) {
+            unlValidatorSigningKeys.add(manifest.signing_key)
+          }
+        })
+      } catch (err) {
+        log.error(`Error fetching UNL for network ${network}`, err)
+      }
+    }
+
+    // Delete old validators that are NOT in the UNL
+    if (unlValidatorSigningKeys.size > 0) {
+      await query('validators')
+        .where('last_ledger_time', '<', oneWeekAgo)
+        .whereNotIn('signing_key', Array.from(unlValidatorSigningKeys))
+        .del()
+    } else {
+      // If no UNL validators found, delete all old validators
+      await query('validators').where('last_ledger_time', '<', oneWeekAgo).del()
```

Comment:
```suggestion
      await query('validators').where('last_ledger_time', '<', oneWeekAgo).whereNull('unl').del()
```

I feel much of the code in this PR can be simplified with this SQL query ^
The `validators` table already has a column titled `unl` which captures this information. Here is the [knexjs docs for `whereNull` operator](https://knexjs.org/guide/query-builder.html#wherenull).

Is this acceptable?